docker-entrypoint.sh

```diff
@@ -50,16 +54,34 @@ check_native_modules() {
                 return 1
             fi
         done
-        
+
         echo "✅ All required native modules ($working_modules) are working correctly"
     else
         echo "ℹ️  No native modules found"
     fi
-    
+
     return 0
 }
 
+# Function to ensure directories exist and have the correct permissions
+create_and_verify_permissions() {
+    mkdir -p $1
 
+    # Only try to set permissions if the current user is root
+    # If the container is run as an unpriveleged user, it means a custom user has been provided to docker and the owner is
+    # responsible for making sure file ownership is correct. Running chown in that case (as non-root) will cause an error
+    if [ "$(id -u)" == "0" ]; then
+        if [ "$(stat -c %u $1)" != "${PUID}" ]; then
+            echo "Fixing $1 ownership..."
+            chown -R ${PUID}:${PGID} $1
+        fi
+    else
+        if [ "$(stat -c %u $1)" != "${id -u}" ]; then
```

Comment:
Very minor change here for edge cases where people specify docker users

```suggestion
        if [ "$(stat -c %u $1)" != "$(id -u)" ]; then
```

Reply:
oops! thanks for catching!